service/history/workflowExecutionUtil.go

```diff
@@ -25,6 +25,24 @@ import (
 	"github.com/uber/cadence/common"
 )
 
+type workflowLockedContext struct {
+	context workflowExecutionContext
+	release releaseWorkflowExecutionFunc
+	err     error
+}
```

Comment:
if the intention is to combine common used structures related to workflow mutation, then plz also add mutable state within this struct, like: https://github.com/uber/cadence/blob/master/service/history/nDCWorkflow.go#L48

so overall probably this struct should looks like this
```go
type workflowLockedContext struct {
	ctx ctx.Context
	context workflowExecutionContext
	mutableState mutableState
	releaseFn releaseWorkflowExecutionFunc	
}
```

Reply:
and please do not include the error within this struct

Reply:
Updated